### PR TITLE
[new release] protocol-9p, protocol-9p-unix and protocol-9p-tool (2.0.2)

### PR DESCRIPTION
**Pull request by djs55**
An implementation of the 9p protocol in pure OCaml

- Project page: <a href="https://github.com/mirage/ocaml-9p">https://github.com/mirage/ocaml-9p</a>
- Documentation: <a href="https://mirage.github.io/ocaml-9p/">https://mirage.github.io/ocaml-9p/</a>

##### CHANGES:

* Update to Cstruct.length in cstruct.6.0.0 and use Fmt.str
  rather than Fmt.strf (@djs55)

### CODE REVIEW

File: packages/protocol-9p-tool/protocol-9p-tool.2.0.2/opam [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  "protocol-9p"
+  "protocol-9p-unix"
+  "base-bytes"
+  "rresult"
+  "logs" {>= "0.5.0"}
+  "fmt"
+  "lambda-term"
+  "win-error"
+  "cmdliner"
+]
```
Comment (avsm) [inline packages/protocol-9p-tool/protocol-9p-tool.2.0.2/opam:20]:
```suggestion
  "cstruct" {>= "6.0.0"}
]
```

Comment (djs55) [reply packages/protocol-9p-tool/protocol-9p-tool.2.0.2/opam:20]: Aha, thanks! Will update

File: packages/protocol-9p-unix/protocol-9p-unix.2.0.2/opam [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  "logs" {with-test & >= "0.5.0"}
+  "win-error"
+  "io-page-unix" {>= "2.0.0"}
+  "ppx_sexp_conv" {< "v0.15"}
+  "alcotest" {with-test & >= "0.4.0"}
+]
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```
Comment (mseri) [inline packages/protocol-9p-unix/protocol-9p-unix.2.0.2/opam:33]:
```suggestion
#  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
```
They seem incompatible with the CI

File: packages/protocol-9p/protocol-9p.2.0.2/opam [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  "base-bytes"
+  "cstruct" {>= "6.0.0"}
+  "cstruct-sexp"
+  "sexplib" {> "113.00.00" & < "v0.15"}
+  "rresult"
+  "mirage-flow" {>= "3.0.0"}
+  "mirage-channel" {>= "4.0.0"}
+  "lwt" {>= "3.0.0"}
+  "astring"
+  "fmt"
```
Comment (mseri) [inline packages/protocol-9p/protocol-9p.2.0.2/opam:20]:
```suggestion
  "fmt" {>= "0.8.7"}
```
Uses `Fmt.kstr`

**Comment by kit-ty-kate** [comment]
Your tests seem to use `/tmp`, which is incompatible with the sandbox. For the future, could you change it to use `$TMPDIR` instead?

**Comment by mseri** [comment]
@kit-ty-kate once
```
2022-02-08 10:28.58: Got resource from pool OCluster
Building on m1-a.ocamllabs.io
fatal: Unable to create '/var/cache/obuilder/ocluster/git/opam-repository.git-b322f7b18d276db68487afc3dfc5a5befdd45818/.git/index.lock': File exists.
```
is fixed, can you retrigger the CI?

**Comment by kit-ty-kate** [comment]
at this point it's easier to push an empty commit or rebase sadly

File: packages/protocol-9p-tool/protocol-9p-tool.2.0.2/opam [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  "protocol-9p-unix"
+  "base-bytes"
+  "rresult"
+  "logs" {>= "0.5.0"}
+  "fmt"
+  "lambda-term"
+  "win-error"
+  "cmdliner"
+  "cstruct" {>= "6.0.0"}
+]
```
Comment (mseri) [inline packages/protocol-9p-tool/protocol-9p-tool.2.0.2/opam:21]:
```suggestion
  "zed" {>= "2.0.0"}
]
```
To fix
```
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.main.eobjs/byte -I /home/opam/.opam/4.08/lib/asetmap -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/camomile -I /home/opam/.opam/4.08/lib/cmdliner -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/cstruct-lwt -I /home/opam/.opam/4.08/lib/cstruct-sexp -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/io-page -I /home/opam/.opam/4.08/lib/io-page-unix -I /home/opam/.opam/4.08/lib/lambda-term -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/lwt -I /home/opam/.opam/4.08/lib/lwt/unix -I /home/opam/.opam/4.08/lib/lwt_log -I /home/opam/.opam/4.08/lib/lwt_log/core -I /home/opam/.opam/4.08/lib/lwt_react -I /home/opam/.opam/4.08/lib/mirage-channel -I /home/opam/.opam/4.08/lib/mirage-flow -I /home/opam/.opam/4.08/lib/mmap -I /home/opam/.opam/4.08/lib/ocaml/threads -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/prometheus -I /home/opam/.opam/4.08/lib/protocol-9p -I /home/opam/.opam/4.08/lib/protocol-9p-unix -I /home/opam/.opam/4.08/lib/react -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/stdlib-shims -I /home/opam/.opam/4.08/lib/win-error -I /home/opam/.opam/4.08/lib/zed -no-alias-deps -o src/.main.eobjs/byte/main.cmo -c -impl src/main.ml)
# File "src/main.ml", line 199, characters 11-23:
# 199 |   inherit [Zed_string.t] LTerm_read_line.term term
#                  ^^^^^^^^^^^^
# Error: Unbound module Zed_string
```

File: packages/protocol-9p-tool/protocol-9p-tool.2.0.2/opam [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+depends: [
+  "ocaml" {>= "4.03.0"}
+  "dune" {>= "1.0"}
+  "protocol-9p"
+  "protocol-9p-unix"
+  "base-bytes"
+  "rresult"
+  "logs" {>= "0.5.0"}
+  "fmt"
+  "lambda-term"
```
Comment (mseri) [inline packages/protocol-9p-tool/protocol-9p-tool.2.0.2/opam:17]:
```suggestion
  "lambda-term" {>= "2.0.0"}
```
alternative fix for
```
# (cd _build/default && /home/opam/.opam/4.08/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.main.eobjs/byte -I /home/opam/.opam/4.08/lib/asetmap -I /home/opam/.opam/4.08/lib/astring -I /home/opam/.opam/4.08/lib/base/caml -I /home/opam/.opam/4.08/lib/bigarray-compat -I /home/opam/.opam/4.08/lib/bytes -I /home/opam/.opam/4.08/lib/camomile -I /home/opam/.opam/4.08/lib/cmdliner -I /home/opam/.opam/4.08/lib/cstruct -I /home/opam/.opam/4.08/lib/cstruct-lwt -I /home/opam/.opam/4.08/lib/cstruct-sexp -I /home/opam/.opam/4.08/lib/fmt -I /home/opam/.opam/4.08/lib/io-page -I /home/opam/.opam/4.08/lib/io-page-unix -I /home/opam/.opam/4.08/lib/lambda-term -I /home/opam/.opam/4.08/lib/logs -I /home/opam/.opam/4.08/lib/lwt -I /home/opam/.opam/4.08/lib/lwt/unix -I /home/opam/.opam/4.08/lib/lwt_log -I /home/opam/.opam/4.08/lib/lwt_log/core -I /home/opam/.opam/4.08/lib/lwt_react -I /home/opam/.opam/4.08/lib/mirage-channel -I /home/opam/.opam/4.08/lib/mirage-flow -I /home/opam/.opam/4.08/lib/mmap -I /home/opam/.opam/4.08/lib/ocaml/threads -I /home/opam/.opam/4.08/lib/parsexp -I /home/opam/.opam/4.08/lib/ppx_sexp_conv/runtime-lib -I /home/opam/.opam/4.08/lib/prometheus -I /home/opam/.opam/4.08/lib/protocol-9p -I /home/opam/.opam/4.08/lib/protocol-9p-unix -I /home/opam/.opam/4.08/lib/react -I /home/opam/.opam/4.08/lib/result -I /home/opam/.opam/4.08/lib/rresult -I /home/opam/.opam/4.08/lib/seq -I /home/opam/.opam/4.08/lib/sexplib -I /home/opam/.opam/4.08/lib/sexplib0 -I /home/opam/.opam/4.08/lib/stdlib-shims -I /home/opam/.opam/4.08/lib/win-error -I /home/opam/.opam/4.08/lib/zed -no-alias-deps -o src/.main.eobjs/byte/main.cmo -c -impl src/main.ml)
# File "src/main.ml", line 199, characters 11-23:
# 199 |   inherit [Zed_string.t] LTerm_read_line.term term
#                  ^^^^^^^^^^^^
# Error: Unbound module Zed_string
```
Probably this one is more correct, not sure

**Comment by mseri** [comment]
All the "cancelled" jobs are either "not installable for dependency conflics" or "success", this can be merged

**Comment by mseri** [comment]
Please @djs55 upstream the changes to the opam files

**Pull request merged**